### PR TITLE
Create writer when there are no records

### PR DESCRIPTION
**Pull request by pcadabam**
Tested this change with IDPC testharness.
@liyinan926 @ibuenros  can you take a look at this?

### CODE REVIEW

**Comment by liyinan926** [comment]
@pcadabam please hold on this until we have a thorough understanding the problems and requirements here.

**Comment by pcadabam** [comment]
@liyinan926 I have updated the pr

File: gobblin-runtime/src/main/java/gobblin/runtime/Fork.java [hunk 1]
```diff
@@ -355,10 +355,25 @@ public void close() throws IOException {
     return new InstrumentedDataWriterDecorator<Object>(writer, this.taskState);
   }
 
+  private void buildWriterIfNotPreset() throws IOException {
```
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/Fork.java]: Can you rename this to `buildWriterIfNotPresent()`?

File: gobblin-runtime/src/main/java/gobblin/runtime/Fork.java [hunk 2]
```diff
@@ -355,10 +355,25 @@ public void close() throws IOException {
...
+      }
+    }
+  }
+
   /**
    * Get new records off the record queue and process them.
    */
   private void processRecords() throws IOException, DataConversionException {
+
+    if (this.taskState.getPropAsBoolean(ConfigurationKeys.WRITER_CREATE_EMPTY_FILE_KEY)) {
```
Comment (liyinan926) [inline gobblin-runtime/src/main/java/gobblin/runtime/Fork.java]: I suggest defining the default (`false`) explicitly in `ConfigurationKeys` and use the default here.

File: gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java [hunk 3]
```diff
@@ -220,6 +220,7 @@
   public static final String WRITER_PRESERVE_FILE_NAME = WRITER_PREFIX + ".preserve.file.name";
   public static final String WRITER_DEFLATE_LEVEL = WRITER_PREFIX + ".deflate.level";
   public static final String WRITER_CODEC_TYPE = WRITER_PREFIX + ".codec.type";
+  public static final String WRITER_CREATE_EMPTY_FILE_KEY = WRITER_PREFIX + ".create.empty.file";
```
Comment (liyinan926) [inline gobblin-api/src/main/java/gobblin/configuration/ConfigurationKeys.java]: I suggest renaming this to `writer.lazy.initialization`

**Comment by pcadabam** [comment]
addressed comments

**Comment by liyinan926** [comment]
LGTM.

**Pull request merged**